include/ica_api.h

```diff
@@ -68,10 +68,38 @@ typedef ica_adapter_handle_t ICA_ADAPTER_HANDLE;
 #define ICA_CBCCS_VARIANT2	2
 #define ICA_CBCCS_VARIANT3	3
 
+/**
+ * ICA flags
+ */
 #define ICA_FLAG_SHW 4 /* static hardware support (symmetric ops - CPACF) */
 #define ICA_FLAG_DHW 2 /* dynamic hardware support (asymmetric ops - CEX) */
 #define ICA_FLAG_SW  1 /* software implementation (fallback / backup) */
 
+/**
+ * ICA properties: key lengths
+ */
+#define ICA_PROPERTY_AES_128		0x00000001
+#define ICA_PROPERTY_AES_192		0x00000002
+#define ICA_PROPERTY_AES_256		0x00000004
+
+#define ICA_PROPERTY_RSA_512		0x00000001
+#define ICA_PROPERTY_RSA_1024		0x00000002
+#define ICA_PROPERTY_RSA_2048		0x00000004
+#define ICA_PROPERTY_RSA_4096		0x00000008
```

Comment:
do we really need this differentiation ? I know there was a limit on RSA for CEX2 but as far as I know even CEX3 supports RSA up to 4096 and we do not have any CEX2 support in the device driver any more.

Reply:
Yes, you are right, today we don't really need it. We could just use ICA_PROPERTY_RSA_ALL. I left the differentiation, because the libica book describes the different RSA key lengths in chapter 5. We may remove the differentiation in the code, but should then also update the book.